certbot/tests/ocsp_test.py

```diff
@@ -150,6 +150,38 @@ def test_revoke(self, mock_ocsp_response, mock_post):
 
         self.assertTrue(revoked)
 
+    @mock.patch('certbot.ocsp.crypto_util.verify_signed_payload')
+    @mock.patch('certbot.ocsp.requests.post')
+    @mock.patch('certbot.ocsp.ocsp.load_der_ocsp_response')
+    def test_responder_signing(self, mock_ocsp_response, mock_post, mock_check):
+        response = _construct_mock_ocsp_response(
+            ocsp_lib.OCSPCertStatus.UNKNOWN, ocsp_lib.OCSPResponseStatus.SUCCESSFUL)
+        mock_ocsp_response.return_value = response
+        mock_post.return_value = mock.Mock(status_code=200)
+
+        issuer = x509.load_pem_x509_certificate(
+            test_util.load_vector('ocsp_issuer_certificate.pem'), default_backend())
+        responder = x509.load_pem_x509_certificate(
+            test_util.load_vector('ocsp_responder_certificate.pem'), default_backend())
+
+        self.checker.ocsp_revoked(self.cert_path, self.chain_path)
+        # On first call, responder and issuer are not the same. Two signatures will be checked
+        # then, first to verify the responder cert (using the issuer public key), second to
+        # to verify the OCSP response itself (using the responder public key).
+        self.assertEqual(mock_check.call_count, 2)
+        self.assertEqual(mock_check.call_args_list[0][0][0].public_numbers(),
+                         issuer.public_key().public_numbers())
+        self.assertEqual(mock_check.call_args_list[1][0][0].public_numbers(),
+                         responder.public_key().public_numbers())
+
+        response.responder_name = issuer.subject
```

Comment:
Although these tests require a bit of boilerplate for setup, it might be beneficial to split the test cases up to different tests while moving the setup to a separate function. (also see the comment about coverage).

Reply:
Done.